### ISSUE TITLE
file disorder

### ISSUE THREAD

**Issue by Baytars** [chrobaktruhlik/spfx-lightbox-image-gallery#6]
The well-named images in my gallery does not seem to be sorted by name. I'm not sure by what they are sorted. Neither can I configure the order in the sidebar. Quite randomly ordered ... 8 - 1 - 2 - 6 - 10 - 12 - 13 - 15. Your README.md gif seems to be of the same issue. I love all the features except this. Could you please fix this! Thank you!
![image](https://user-images.githubusercontent.com/20458079/221375673-e27c4bdb-9b4c-4ac2-a1ae-6d95fecf5934.png)

**Comment by chrobaktruhlik** [comment]
Gif in README.md is from original repository. Someday, over time, I will create my own.

Release 1.0.1 - order logic (SharePoint library column):
- Root Document Library: by Name (internal name when library was created)
- Folders in Library: by Name (only)
- Files in Folder: by TimeCreated 

In next release will be fix to:
- Root Document Library: by Title (own name instead of internal name)
- Folders in Library: choice: Name, Name reverse, TimeCreated, TimeCreated reverse
- Files in Folder: by TimeCreated

In my test environment (SharePoint Online - Modern Pages) it sorts correctly.

**Comment by Baytars** [comment]
When is the next release scheduled?

**Comment by Baytars** [comment]
> Files in Folder: by TimeCreated

Please offer the choices as you do for the folder. The files need to be sorted by name as well. My pictures' names are patterned, not randomly assigned.

**Comment by Baytars** [comment]
Also, please allow surfacing all pictures in subfolders (flatening subfolders to make all pictures look like coming from the same directory) so that I don't have to manually click through subfolders to view pictures in a different subfolder. The built-in Image Gallery webpart can already do the work. It is a pity that it has a maximum limit of 255 pictures, and does not support viewing in full screen. In addition, while the first bunch of pictures seem to be arranged correctly, the next subfolder seems to be a bit disordered too. I don't know what the order logic is.
Implementing the desired order logic may be a little complex as my pictures are arranged as below:
```
c001/
    001.jpg
    002.jpg
    003.jpg
c002/
    001.jpg
    002.jpg
    003.jpg
```
My desired order will be `c001/001.jpg` - `c001/002.jpg` - `c001/003.jpg` - `c002/001.jpg` - `c002/002.jpg` - `c002/003.jpg` (sort by full path). A typical behavior of Highlighted Content webpart will be in this order: `c001/001.jpg` - `c002/001.jpg` - `c001/002.jpg` - `c002/002.jpg` - `c001/003.jpg` - `c002/003.jpg`, not what is desired, too.

**Comment by chrobaktruhlik** [comment]
I started to adopt the original repository because a lot of things are not as I would have imagined. For example the sorting of folders/files you describe. The compiled release 1.0.1 is almost identical to the original. It retains the version numbering continuing from the original. Release 2.0.0 will be a new, modified version. When? Sometime in March 2023.

I don't plan to display all images in all subfolders of the folder as you describe. Just as it is. Only the images of the current folder will be displayed.

However, thank you for the idea for the new feature.

**Comment by chrobaktruhlik** [comment]
Partially resolved in 2.0.0.
New feature: "flatening subfolders to make all pictures look like coming from the same directory" is being planned in a future version.

**Comment by chrobaktruhlik** [comment]
Partially resolved in 2.0.0.
New feature: "flatening subfolders to make all pictures look like coming from the same directory" is being planned in a future version.
